libs/core/coroutines/src/detail/coroutine_impl.cpp

```diff
@@ -96,4 +97,52 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
         // should not get here, never
         HPX_ASSERT(this->m_state == super_type::ctx_running);
     }
+
+    // execute the coroutine function directly in the context of the calling
+    // thread
+    coroutine_impl::result_type coroutine_impl::invoke_directly(
+        coroutine_impl::arg_type arg)
+    {
+        using context_exit_status = super_type::context_exit_status;
```

Comment:
The only HPX thread specifics would be suspension (which suspends the executing (HPX) thread), but since the execution agent handles suspension generically, we should be able to execute things directly on any thread.

I mostly kept the execution here as similar to the 'normal' thread execution for now. We might be able to remove things, however.